### PR TITLE
DurationFormatter: Digital Data Provider

### PR DESCRIPTION
**Pull request by kartva**
- add `DigitalDurationDataV1` data struct to hold digital duration formatter information.

### CODE REVIEW

**Comment by DesmondWillowbrook** [comment]
@younies ready for review

File: components/experimental/src/duration/provider/digital.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use displaydoc::Display;
+use icu_provider::prelude::*;
+
+#[icu_provider::data_struct(DigitalDurationDataV1Marker = "duration/digital@1")]
+#[derive(Debug, Copy, Clone, PartialEq)]
+#[cfg_attr(
```
Comment (younies) [inline components/experimental/src/duration/provider/digital.rs:17]: This should be something like this 

```rust
pub struct DigitalDurationDataV1 {
    pub separator: Cow<'data, str>,

    pub padding: u8,
}
```

File: components/experimental/src/duration/provider/digital.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+/// Represents possible variants of the hour-minute pattern.
+pub enum HmVariant {
+    /// hh:mm
+    DoubleHourColonSep = 0,
+    /// h:mm
+    SingleHourColonSep = 1,
+    /// h.mm
+    SingleHourDotSep = 2,
+}
+
```
Comment (younies) [inline components/experimental/src/duration/provider/digital.rs:54]: we do not need this

File: components/experimental/src/duration/provider/digital.rs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "hh:mm:ss" => Ok(HmsVariant::DoubleHourColonSep),
+            "h:mm:ss" => Ok(HmsVariant::SingleHourColonSep),
+            "h.mm.ss" => Ok(HmsVariant::SingleHourDotSep),
+            _ => Err(UnknownPatternError(s.to_string())),
+        }
+    }
+}
```
Comment (younies) [inline components/experimental/src/duration/provider/digital.rs:113]: You can implement `FromStr`  for  `DigitalDurationDataV1`

example:

```rust
 match s {
            "hh:mm:ss" => Ok(DigitalDurationDataV1{separator: ":", padding: 2}),
            ... etc.
        }
```

also, I think there is a pattern --> `"h,mm,ss"`

**Review by younies — CHANGES_REQUESTED** [review]
Good work,

I have added for you some comments to follow.

You can send to me the provider in a separate PR first, and then we can go through the implementation.

File: components/experimental/src/duration/provider/digital.rs (hunk 1 above)
Comment (kartva) [reply components/experimental/src/duration/provider/digital.rs:17]: To clarify, `padding` here refers to hour padding? Do we pick up one separator and derive hm, hms, and ms representations like the following:
`separator: ",", padding = 2 => hm: "hh,mm", hms: "hh,mm,ss", ms: "mm,ss"`?

**Pull request closed without merging**